src/components/manage/Controlpanels/Controlpanel.jsx

```diff
@@ -141,6 +141,35 @@ class Controlpanel extends Component {
   }
   form = React.createRef();
 
+  // Goal would be to replace the Form schema prop in line 182, with this function
+  filterSchema = (controlpanel) => {
+    const panelType = controlpanel['@id'].replace('/@controlpanels/', '');
+    let unwantedSettings = [];
+
+    // Allows to specify fields to remove per controlpanel
+    switch (panelType) {
+      case 'language':
+        unwantedSettings = ['display_flags', 'always_show_selector'];
+        break;
+      default:
+        unwantedSettings = ['none!'];
+    }
```

Comment:
This is fine. But the way I would usually expect to see this written in Javascript is to build a nested object:
```
const unwantedSettings = {
  language: ['display_flags', 'always_show_selector'],
};
```

And then use `unwantedSettings[panelType] || []` below